### DISCUSSION TITLE
Returns list with join between tables done or returns raw table and use parent to do another query?

### DISCUSSION THREAD

**Discussion by gabrielferreiraa** [apollographql/apollo-server#7083] [Q&A]
Returns list with join between tables done or returns raw table and use id field (parent) to do another query? e.g:

[1]
![carbon (1)](https://user-images.githubusercontent.com/17201177/198093486-7a3006c8-c1a5-4d7a-b085-b29545fd11b6.png)

or

[2]
![carbon (2)](https://user-images.githubusercontent.com/17201177/198093540-a14c7880-1614-4ced-9885-032b45bef359.png)

ref: https://twitter.com/ggabrieo/status/1585319686245584896

**Comment by trevor-scheer** [comment]
The example here isn't quite correct - in [2] you have a `posts` field which should be on the `User` type rather than the `Query` type.

As your schema scales, fetching all possible data that a user might have will become problematic. For that reason, you generally only want to resolve fields as needed.

Note that resolving users and then each user's posts individually can be problematic as well! This is a commonly encountered problem (named N+1, since you have your first query followed by N queries for posts where N is the number of users you got in the first query). I recommend looking into DataLoader and searching for information on "GraphQL N+1" if you're not already familiar with the topic.
